### PR TITLE
Toyota: use torque control for Japanese-made 2023 RAV4

### PR DESCRIPTION
**Pull request by sshane**

### CODE REVIEW

**Comment by sshane** [comment]
Pending a few routes, is this how you would do it @adeebshihadeh, or split it up into 4 platforms for the ICE and hybrid?

File: selfdrive/car/toyota/interface.py [hunk 1]
```diff
@@ -28,8 +28,14 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
       ret.safetyConfigs[0].safetyParam |= Panda.FLAG_TOYOTA_ALT_BRAKE
 
     if candidate in ANGLE_CONTROL_CAR:
-      ret.dashcamOnly = True
       ret.steerControlType = SteerControlType.angle
+      for fw in car_fw:
+        # Detect if we can control the car with torque (cars made in Japan)
+        if fw.ecu == "eps" and not fw.fwVersion == b'8965B42371\x00\x00\x00\x00\x00\x00':
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/interface.py:34]: this blacklist seems too specific? I doubt all 23 Rav4's with angle will have this FW, but they can potentially fingerprint as torque cars now

**Review by adeebshihadeh — APPROVED** [review]

File: selfdrive/car/toyota/interface.py (hunk 1 above)
Comment (sshane) [reply selfdrive/car/toyota/interface.py]: I should not have placed that `not` there, this is the FW for the torque cars.

File: selfdrive/car/toyota/interface.py [hunk 2]
```diff
@@ -28,8 +28,14 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
...
+          ret.steerControlType = SteerControlType.torque
+
+    if ret.steerControlType == SteerControlType.angle:
```
[the remark this answers is not in the document]
Comment (sshane) [reply selfdrive/car/toyota/interface.py:37]: It should be more clear now. I wanted to explicitly make the default be angle control if we fingerprint a car in the set. then whitelist certain FW.

[the remark this answers is not in the document]
Comment (sshane) [reply selfdrive/car/toyota/interface.py]: Made it one if statement, but reads more confusingly now. Let me know if you had a better idea

**Pull request merged**